src/txmempool.cpp

```diff
@@ -108,18 +122,23 @@ void CTxMemPool::UpdateForDescendants(txiter update_it, cacheMap& cache, const s
     int64_t modify_size = 0;
     CAmount modify_fee = 0;
     int64_t modify_count = 0;
+    // All entries in the update_cache have
+    // already been checked against excluded list
     for (txiter child_it : update_cache) {
         const CTxMemPoolEntry& child = *child_it;
-        if (!exclude.count(child.GetTx().GetHash())) {
-            modify_size += child.GetTxSize();
-            modify_fee += child.GetModifiedFee();
-            modify_count++;
-            mapTx.modify(child_it, update_ancestor_state(update_it->GetTxSize(), update_it->GetModifiedFee(), 1, update_it->GetSigOpCost()));
-        }
+        modify_size += child.GetTxSize();
```

Comment:
Github won't let me leave a comment up at line 106, so I'm leaving it here.  I think the change in this commit of trying to not include things in `exclude` is broken, because it's possible you'll have a cache entry hit for something that is excluded, and then because things that are in the cache get swapped at line 106 to an entry in the vector that `already_traversed` is advanced past, you'll never pop it before you get down to this loop.

I tend to think that this improvement to leaving things out from exclude isn't worth it; seems like quite a bit of added complexity to this logic without an obvious performance improvement.

Reply:
We hold the invariant that everything in the cache is not excluded? So how would this have a bug?

Reply:
Ah I see what you're saying. See comment above, will fix.